cmd/swapd/main.go

```diff
@@ -338,11 +327,24 @@ func (d *daemon) make(c *cli.Context) error {
 		}
 	}
 
+	ethEndpoint := common.DefaultEthEndpoint
+	if c.String(flagEthereumEndpoint) != "" {
+		ethEndpoint = c.String(flagEthereumEndpoint)
+	}
+	ec, err := ethclient.Dial(ethEndpoint)
+	if err != nil {
+		return err
+	}
+	chainID, err := ec.ChainID(d.ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
Moved the above code above from `newBackend`  so we would have access to the Chain ID sooner.